### DISCUSSION TITLE
BitNet 1.58b models reproduced successfully

### DISCUSSION THREAD

**Discussion by kalomaze** [ggml-org/llama.cpp#6384] [General]
https://huggingface.co/1bitLLM/bitnet_b1_58-3B

**Comment by Dampfinchen** [comment]
great results. But why is it so big? 13 GB seems way too much for a 1.58 Bit 3B model.

**Reply by netrunnereve** [reply]
AFAIK the weights are stored as {-1, 0, 1} but in f32.

**Reply by Dampfinchen** [reply]
I see. Would it be possible to store it in FP16 or even 4 Bit without a noticeable dip in quality? If not, it would take a lot of disk space for 7B models and up. But if that's the compromise of having 1.58 Bit models I think most people will gladly take it.

**Reply by netrunnereve** [reply]
Yeah you could store it however you like, even in ternary if you pack it properly. They likely use f32 as all the ML libraries support it and they don't have to write special code for inference. Optimizing that into something that's faster and more compact is our job :wink:

**Reply by TomLucidor** [reply]
Will this have a 5x speed boost? What about the new Sherry quantization method?

**Comment by EriIaz** [comment]
> great results. But why is it so big? 13 GB seems way too much for a 1.58 Bit 3B model.

Isn't it FP32?

**Comment by hrstoyanov** [comment]
Are there any plans for efficient non-gpu (AVX) implementation as part of llama.cpp?
